### DISCUSSION TITLE
file format of memory card/where does the emulator save the games?

### DISCUSSION THREAD

**Discussion by Jonathan-Sutjandra** [kootstra-rene/enge-js#51] [Q&A] [answered]
When attempting to put a epsxe000.mcr as a memory file into the emulator by dragging and dropping or using the upload buttons, the game I am emulating says that the memory card has not been formatted. On a different device, I did not use a memory card file, and when closing then restarting the emulator, my save file was still there. Since booting up a .mcr seems to not work but saving without a card does, where does the emulator save the game? (The game I was emulating was "Metal Gear Solid.")

**Comment by kootstra-rene** [comment, marked as answer]
Hi, the memory card is stored in the browser local storage. There is a story to implement memory card management for uploading and downloading memory cards. Sadly I have limited time to build it.

However loading mem cards should work. Can you attach the mem card file to this issue so that I can test an fix it.

**Reply by Jonathan-Sutjandra** [reply]
Thank you for the information. I tried a different mem card and it worked, and also found that the mem card I downloaded the file from was corrupted in the first place, so that's that. Sorry for that and thank you again.

**Reply by kootstra-rene** [reply]
No problem and your welcome.
